recipes/ms-gsl/meta.yaml

```diff
@@ -0,0 +1,46 @@
+{% set name = "ms-gsl" %}
+{% set version = "4.0.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/microsoft/GSL/archive/v{{ version }}.tar.gz
+  sha256: f0e32cb10654fea91ad56bde89170d78cfbf4363ee0b01d8f097de2ba49f6ce9
+
+build:
+  number: 0
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('cxx') }}
+    - ninja
+
+test:
+  requires:
+    - cmake
+    - {{ compiler('cxx') }}
+    - ninja
+  files:
+    - test
+
+  commands:
```

Comment:
It's encouraged to add some tests checking that the expected headers, libraries, executables are present in the package: https://github.com/conda-forge/zeromq-feedstock/blob/main/recipe/meta.yaml#L46-L64 . And if you are shipping headers, a `build/run_exports` is encouraged.

(P.S. I'm not an official reviewer, so above is just an observation.)

Reply:
The rest subdirectory of the recipe contains a fully formed program that uses the library and compiled against it.

Not sure I follow the run_exports bit, can you explain?

For context, this is a header only library and has no dependencies.

Reply:
`run_exports` is a way for library recipe developers to impose that if a downstream package compiles against this one, that package should have this one in its requirements/run with certain constraints. https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#export-runtime-requirements But if this is header-only, run_exports is irrelevant.

I agree, your test build largely assures the package is functioning. But I suspect the reviewers may like file checks anyways. Here's a header-only pattern. https://github.com/conda-forge/eigen-feedstock/blob/main/recipe/meta.yaml#L27-L30